flowgraph/core/record.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Record flow graphs by running Python code.
+
+This module is a convenient entry point for users of the package, wrapping
+the `FlowGraphBuilder` and `Tracer` into easy-to-use functions.
+"""
+from __future__ import absolute_import
+
+from traitlets import HasTraits, Instance
+
+from flowgraph.trace.tracer import Tracer
+from .flow_graph_builder import FlowGraphBuilder
+
+
+class Recorder(HasTraits):
+    """ Context manager to record flow graphs.
+
+    Like all context managers, this class should be used in a `with` statement::
+
+        with Recorder() as graph:
+            [...]
```

Comment:
If it's not too much trouble, add one example on what type of operations can be done on `graph` (instead of/in addition to the generic `[...]` OR (since looking at the test cases this may not be trivial) briefly explain the usage pattern.